src/main/java/core/basesyntax/storage/Storage.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.storage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private final Map<String, Integer> inventory = new HashMap<>();
+
+    public void addFruit(String fruit, int quantity) {
+        inventory.merge(fruit, quantity, Integer::sum);
+    }
+
+    public void removeFruit(String fruit, int quantity) {
+        inventory.merge(fruit, -quantity, Integer::sum);
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.

Before updating the inventory, you should check if the resulting quantity will not be negative. If it is, you should throw a `RuntimeException` to prevent invalid states in the storage.